bot/exts/valentines/be_my_valentine.py

```diff
@@ -118,56 +109,41 @@ async def send_valentine(
     @commands.cooldown(1, 1800, BucketType.user)
     @send_valentine.command(name='secret')
     async def anonymous(
-        self, ctx: commands.Context, user: Optional[discord.Member] = None, *, valentine_type: str = None
+        self, ctx: commands.Context, user: discord.Member, *, valentine_type: str = None
     ) -> None:
         """
         Send an anonymous Valentine via DM to to a user, if specified, or to a random with the lovefest role.
 
-        **This command should be DMed to the bot.**
-
-        syntax : .bemyvalentine secret [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        syntax : .bemyvalentine secret [user] [p/poem/c/compliment/or you can type your own valentine message]
         (optional)
 
-        example : .bemyvalentine secret (sends valentine as a poem or a compliment to a random user in DM making you
-        anonymous)
         example : .bemyvalentine secret Iceman#6508 p (sends a poem to Iceman in DM making you anonymous)
         example : .bemyvalentine secret Iceman#6508 Hey I love you, wanna hang around ? (sends the custom message to
         Iceman in DM making you anonymous)
         """
-        if ctx.guild is not None:
-            # This command is only DM specific
-            msg = "You are not supposed to use this command in the server, DM the command to the bot."
-            return await ctx.send(msg)
-
-        if user:
-            if Lovefest.role_id not in [role.id for role in user.roles]:
-                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
-                return await ctx.send(message)
+        if Lovefest.role_id not in [role.id for role in user.roles]:
+            message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+            await ctx.message.delete()
```

Comment:
I don't think we should delete the message that invokes the command. Have the command that caused the error s good reference for the user, to see where they went wrong.

Reply:
I deleted it so that it still acts like a "Secret", and confirmation are sent to the user on DMs, so the user always has reference.

Reply:
That makes sense in this context, agreed.